script/install/debootstrap.bash

```diff
@@ -173,5 +173,47 @@ echo "$chroot_cmd update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9
 chroot "$ISOLATE_ROOT" update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 75
 # gcc 9 done
 
-# let user know that chroot installs are finished
+[ -z "$TRAVIS" ] && bash script/confirm.bash 'Install .Net Core (C#)' && {
+  set -x
+
+  : Installing .Net Core for C#
+  # https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-ubuntu-1604
+
+  : Downloading package info from Microsoft
+  UBUNTU_VERSION=$(lsb_release --release --short)
+  chroot "$ISOLATE_ROOT" wget -q "https://packages.microsoft.com/config/ubuntu/${UBUNTU_VERSION}/packages-microsoft-prod.deb" -O packages-microsoft-prod.deb
+  chroot "$ISOLATE_ROOT" dpkg -i packages-microsoft-prod.deb
+  chroot "$ISOLATE_ROOT" rm packages-microsoft-prod.deb
+  [ "$UBUNTU_VERSION" == "18.04" ] && chroot "$ISOLATE_ROOT" sudo add-apt-repository universe
+  chroot "$ISOLATE_ROOT" apt-get install -y apt-transport-https
+  chroot "$ISOLATE_ROOT" apt-get update
+
+  : Installing dotnet 3.1
+  chroot "$ISOLATE_ROOT" apt-get install -y dotnet-sdk-3.1
+
+  : Creating runtime config file
+  # The dotnet command will not execute the compiled .exe unless there is a config file
+  # which tells it what runtime framework to use.
+  # https://stackoverflow.com/questions/46065777/is-it-possible-to-compile-a-single-c-sharp-code-file-with-the-net-core-roslyn-c
+
+  RUNTIME_CONFIG_PATH="$ISOLATE_ROOT/usr/share/dotnet-config.json"  # used in the compile command
+  RUNTIME_CONFIG_TEMPL="script/csharp/dotnet-config-template.json"
 
+  # referenced in RUNTIME_CONFIG_TEMPL file
+  DOTNET_FW_VERSION=$(dotnet --version | sed -r "s/([0-9].[0-9].[0-9])[0-9]*/\1/") \
```

Comment:
This is trying to get the .NET Core Runtime version from the .NET Core SDK version, right? There are examples in https://docs.microsoft.com/en-us/dotnet/core/versions/ where the third components don't match. I think the correct way to get the runtime version would be to parse the output of [dotnet --list-runtimes](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet?tabs=netcore21), however the [Runtime Binding Behaviour design doc](https://github.com/dotnet/designs/blob/master/accepted/runtime-binding.md) recommends setting the patch number in runtimeconfig.json to 0 (which is much easier!).
```
  # Compute the .NET Core Runtime version from the .NET Core SDK version
  # The major and minor versions are the same (https://docs.microsoft.com/en-us/dotnet/core/versions/).
  # The patch version should be set to 0 (https://github.com/dotnet/designs/blob/master/accepted/runtime-binding.md).
  DOTNET_FW_VERSION=$(chroot "$ISOLATE_ROOT" dotnet --version | sed 's/\.[0-9]\+$/.0/') \
    ...
```
(If we wanted to simplify things further we could be sneaky and use the runtimeconfig of dotnet.dll itself, /usr/share/dotnet/sdk/3.1.101/dotnet.runtimeconfig.json, but that feels fragile.)

Reply:
I figured it probably wasn't reliable to assume the output of `dotnet --list-runtimes` will be consistent across versions. Good find with the 0 patch version